docs/cloudevents/cloudevents-spans.md

```diff
@@ -35,168 +31,26 @@ consult the
-|          Trace 1                                      |
-|                                                       |
-|          +---------------------------------------+    |
-|    +---> | Create event                          |    |
-|    |     +---------------------------------------+    |
-|    |     +---------------------------------------+    |
-|    |     | Send (auto-instr)                     |    |
-|    |     +---------------------------------------+    |
-|    |        +------------------------------------+    |
-|    |        | Intermediary: Received (auto-instr)|    |
-|    |        +------------------------------------+    |
-|    |        +------------------------------------+    |
-|    |        | Intermediary: Send (auto-instr)    |    |
-|    |Link    +------------------------------------+    |
-|    |                                                  |
-|    |                                                  |
-|    |                                                  |
-|    |     Trace 2                                      |
-|    |                                                  |
-|    |     +---------------------------------------+    |
-|    |     | Consumer: Receive (auto-instr)        |    |
-|    |     +---------------------------------------+    |
-|    |       +-------------------------------------+    |
-|    +------ | Consumer: Process                   |    |
-|            +-------------------------------------+    |
-|                                                       |
-+-------------------------------------------------------+
-```
-
 ## Conventions
 
-To achieve the trace above, it is necessary to capture the context of
-the event creation so that when the CloudEvent reaches its destination(s), this
-context can be continued. Each CloudEvent acts then as the medium of this
-context propagation.
-
-This document relies on the CloudEvents specification, which defines this
-context propagation mechanism via the
-[CloudEvents Distributed Tracing Extension](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/extensions/distributed-tracing.md).
-Once the trace context is set on the event
-via the Distributed Tracing Extension, it MUST not be modified.
-
-The remainder of this section describes the semantic conventions for Spans
-required to achieve the proposed trace.
-
-### Spans
+CloudEvents can be sent via various communication protocols. When the protocol
+used to send CloudEvents has a well established support for trace context propagation,
+instrumentations SHOULD propagate the context using this built-in mechanism.
 
-#### Creation
-
-Instrumentation SHOULD create a new span and populate the
+If CloudEvents are instrumented independently of the protocol used,
+CloudEvent-specific instrumentations SHOULD use the
 [CloudEvents Distributed Tracing Extension](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/extensions/distributed-tracing.md)
```

Comment:
```suggestion
The CloudEvents specification defines a context propagation mechanism via CloudEvents in the [CloudEvents Distributed Tracing Extension](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/extensions/distributed-tracing.md). This mechanism satisfies [context propagation requirements of OpenTelemetry messaging semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-spans.md#context-propagation).
```

I'd avoid SHOULD wording, and just hint at the existence of the context propagation extension.